src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchQueryHelper.java

```diff
@@ -0,0 +1,166 @@
+        return sortNode;
+    }
+
+    private void addCaseTypeSorts(String caseTypeId, UseCase useCase, ArrayNode sortNode) {
+        Optional<CaseTypeDefinition> caseTypeOpt = getCaseTypeOperation.execute(caseTypeId, CAN_READ);
+        caseTypeOpt.ifPresent(caseType -> searchQueryOperation.getSortOrders(caseType, useCase)
+            .forEach(field -> sortNode.add(buildSortOrderFieldNode(caseType, field))));
+    }
+
+    private ObjectNode buildSortOrderFieldNode(CaseTypeDefinition caseTypeDefinition, SortOrderField sortOrderField) {
+        ObjectNode objectNode = objectMapper.createObjectNode();
+        final CommonField commonField = caseTypeDefinition.getCommonFieldByPath(sortOrderField.getCaseFieldId()).orElseThrow(() ->
+            new NullPointerException(String.format("Case field '%s' does not exist in configuration for case type '%s'.",
+                sortOrderField.getCaseFieldId(), caseTypeDefinition.getId()))
+        );
+        final FieldTypeDefinition fieldType = commonField.getFieldTypeDefinition();
+
+        StringBuilder sb = new StringBuilder();
+
+        if (sortOrderField.isMetadata()) {
+            sb.append(MetaData.CaseField.valueOfReference(sortOrderField.getCaseFieldId()).getDbColumnName());
+        } else {
+            sb.append(CASE_DATA_PREFIX).append(sortOrderField.getCaseFieldId());
+            if (fieldType.getType().equals(FieldTypeDefinition.COLLECTION)) {
+                sb.append(COLLECTION_VALUE_SUFFIX);
+            }
+        }
+
+        if ((sortOrderField.isMetadata() && elasticsearchMappings.isDefaultTextMetadata(sb.toString()))
+             || (!sortOrderField.isMetadata() && elasticsearchMappings.isDefaultTextCaseData(fieldType))) {
+            sb.append(KEYWORD_SUFFIX);
+        }
+
+        objectNode.set(sb.toString(), new TextNode(sortOrderField.getDirection()));
+        return objectNode;
+    }
+
+    private JsonNode stringToJsonNode(String jsonSearchRequest) {
+        return objectMapperService.convertStringToObject(jsonSearchRequest, JsonNode.class);
+    }
+
+    private void rejectBlackListedQuery(String jsonSearchRequest) {
+        List<String> blackListedQueries = applicationParams.getSearchBlackList();
+        Optional<String> blackListedQueryOpt = blackListedQueries
+            .stream()
+            .filter(blacklisted -> {
+                Pattern p = Pattern.compile("\\b" + blacklisted + "\\b");
+                Matcher m = p.matcher(jsonSearchRequest);
+                return m.find();
+            })
+            .findFirst();
+        blackListedQueryOpt.ifPresent(blacklisted -> {
+            throw new BadSearchRequest(String.format("Query of type '%s' is not allowed", blacklisted));
+        });
+    }
+
+    private List<String> buildCaseTypeIds(List<String> caseTypeIds) {
+        return CollectionUtils.isEmpty(caseTypeIds)
+            ? userService.getUserCaseTypes().stream().map(CaseTypeDefinition::getId).collect(Collectors.toList())
```

Comment:
temporary: now that we are making ctid optional in the external search API to allow searching on all existing case types, is there any benefit in doing this?

Reply:
furthermore, this can negatively impact search performance. If you consider a just redeployed DataStore, where all caches are empty. For each user jurisdiction we need to load its definition from def store before we can make the search. This might be too inefficient. It heavily relies on caching of the definitions.